frontend/src/concepts/pipelines/apiHooks/useArtifactStorage.ts

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react';
+
+import { SupportedArea, useIsAreaAvailable } from '~/concepts/areas';
+import { fetchStorageObject, fetchStorageObjectSize } from '~/services/storageService';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { Artifact } from '~/third_party/mlmd';
+import {
+  extractS3UriComponents,
+  getArtifactUrlFromUri,
+} from '~/concepts/pipelines/content/artifacts/utils';
+import { useGetArtifactById } from './useGetArtifactById';
+
+export const useArtifactStorage = (
+  artifact?: Artifact,
+): {
+  artifactFeatureFlagEnabled: boolean;
+  objectUrl: string | undefined;
+  objectSize: number | undefined;
+  downloadedObject: string | undefined;
+} => {
+  const s3EndpointAvailable = useIsAreaAvailable(SupportedArea.S3_ENDPOINT).status;
+  const artifactApiAvailable = useIsAreaAvailable(SupportedArea.ARTIFACT_API).status;
+  const { namespace } = usePipelinesAPI();
+
+  const path = artifact?.getUri();
+  const artifactId = artifact?.getId().toString();
+
+  const [downloadedObject, setDownloadedObject] = useState<string | undefined>(undefined);
+  const [objectUrl, setStorageUrl] = useState<string | undefined>(undefined);
+  const [objectSize, setObjectSize] = useState<number | undefined>(undefined);
```

Comment:
Could be simplified:
```suggestion
  const [downloadedObject, setDownloadedObject] = useState<string>();
  const [objectUrl, setStorageUrl] = useState<string>();
  const [objectSize, setObjectSize] = useState<number>();
```